driver-core/src/test/java/com/datastax/driver/core/CCMBridge.java

```diff
@@ -1282,12 +1295,48 @@ private String randomizePorts(CharSequence str) {
       Matcher matcher = RANDOM_PORT_PATTERN.matcher(str);
       StringBuffer sb = new StringBuffer();
       while (matcher.find()) {
-        matcher.appendReplacement(sb, Integer.toString(TestUtils.findAvailablePort()));
+        matcher.appendReplacement(sb, Integer.toString(selectAvailablePort()));
       }
       matcher.appendTail(sb);
       return sb.toString();
     }
 
+    // Collects port numbers from configuration map. May catch some non-issue false positives.
+    private static Collection<Integer> collectPorts(Map<String, Object> config) {
+      HashSet<Integer> set = new HashSet<Integer>();
+      for (Object value : config.values()) {
+        try {
+          int result = Integer.parseInt(value.toString());
+          if ((1 <= result) && (result <= 65535)) {
+            set.add(result);
+          }
+        } catch (NumberFormatException e) {
+        }
+      }
+      return set;
+    }
+
+    private int findUnselectedPort() throws RuntimeException {
```

Comment:
this throws RuntimeException shouldn't be needed, is it?

Reply:
Right, it's erased now.